### PR TITLE
User Account Enhancements

### PR DESCRIPTION
**Pull request by cesarvarela**
Part of #1783 

It should be available here: https://deploy-preview-46--cesarvarela-staging.netlify.app/

### CODE REVIEW

**Comment by pdcp1** [comment]
credentials: `mail@cesarvarela.com:12345678`

**Review by pdcp1 — CHANGES_REQUESTED** [review]

File: site/gatsby-site/src/components/users/UserEditModal.js [hunk 1]
```diff
@@ -97,12 +115,14 @@ export default function UserEditModal({ onClose, userId }) {
...
-                  </div>
-                </details>
+                {isRole('admin') && (
+                  <details>
+                    <summary className="cursor-pointer text-sm">Supported Roles</summary>
+                    <div className="mt-2">
+                      <RolesTable roles={supportedRoles} />
+                    </div>
+                  </details>
+                )}
```
[the remark this answers is not in the document]
Comment (cesarvarela) [reply site/gatsby-site/src/components/users/UserEditModal.js:125]: `subscriber@cesarvarela.com:12345678`

[the remark this answers is not in the document]
Comment (pdcp1) [reply site/gatsby-site/src/components/users/UserEditModal.js:125]: It looks good 👍

File: site/gatsby-site/src/components/forms/SubmitForm.js [hunk 2]
```diff
@@ -100,24 +101,28 @@ const SubmitForm = () => {
...
-    if (queryParams.image_url) {
-      queryParams.cloudinary_id = getCloudinaryPublicID(queryParams.image_url);
-    }
+      if (user?.profile?.email) {
+        submission.user = { link: user.id };
 
-    setSubmission(queryParams);
-  }, []);
+        if (user.customData.first_name && user.customData.last_name) {
+          submission.submitters = [`${user.customData.first_name} ${user.customData.last_name}`];
```
[the remark this answers is not in the document]
Comment (cesarvarela) [reply site/gatsby-site/src/components/forms/SubmitForm.js:119]: Yup, it is exactly that.

[the remark this answers is not in the document]
Comment (cesarvarela) [reply site/gatsby-site/src/components/forms/SubmitForm.js:119]: `submitters` field accomplishes two taks:
- allow anonymous users set a submitter field
- have a name associated to a submission/report when exporting the database (we could fix it with a custom script but it is way more work)

[the remark this answers is not in the document]
Comment (pdcp1) [reply site/gatsby-site/src/components/forms/SubmitForm.js:119]: Got it

**Review by pdcp1 — APPROVED** [review]

**Comment by cesarvarela** [comment]
@smcgregor if you want to check this out before merging, https://deploy-preview-46--cesarvarela-staging.netlify.app/

`mail@cesarvarela.com:12345678`

**Comment by smcgregor** [comment]
Looking good in general, but I have questions/comments.

* How would multiple submitters work?
* What happens if I edit the submitter after it is in the review queue?
* The user table is not in the snapshots by design, but I believe this changeset stores user-presented data in the user table. We need a snapshot strategy that is aware of privacy and security concerns here.
* This needs a migration to associate all the historical submissions with the user accounts that submitted them. We can limit this to me, @LKchemposer, @Janetbananet, Danny, and @kateperk at present.
* This text is no longer valid: `Enter your name here. If more than one person contributed to collecting the incident, press “enter” to begin a new item. Alternatively, you can enter "Anonymous" as submitter if you do not want to be attributed in the database. Entries in this field will be added to the public leaderboard.`
* The review form renders twice before it is stable

**Comment by cesarvarela** [comment]
> How would multiple submitters work?

`submissions` and `reports`  have a `user` field containing the user id of the account logged in.

`submissions` and `reports` have a `submitters` field that contains a list of names `[ 'John Doe', 'Jane Doe']`. The first name of this list comes from the name entered in the associated user account, but it is not enforced in the DB (editing your account details won't update the `submitters` field at the moment). 

> What happens if I edit the submitter after it is in the review queue?

You can change the names in the `submitters` list, but the user account associated with the `submission` can't be modified.

> The user table is not in the snapshots by design, but I believe this changeset stores user-presented data in the user table. We need a snapshot strategy that is aware of privacy and security concerns here.

If we keep the `submitters` field as it is, we don't need to expose the `users` collection.

> This needs a migration to associate all the historical submissions with the user accounts that submitted them. We can limit this to me, @LKchemposer, @Janetbananet, Danny, and @kateperk at present.

I'll need a list of the first and last names, user ids, and emails as they are shown on the user admin table.

> This text is no longer valid: Enter your name here. If more than one person contributed to collecting the incident, press “enter” to begin a new item. Alternatively, you can enter "Anonymous" as submitter if you do not want to be attributed in the database. Entries in this field will be added to the public leaderboard.

Could you help me with that?

> The review form renders twice before it is stable

ok


@smcgregor

**Comment by smcgregor** [comment]
When the user is logged in: "You are currently logged in so this submission will automatically be associated with your account. If you would like to remain anonymous, please open a private browsing window to submit."

New issues

* `editing your account details won't update the submitters field at the moment`
* `You can change the names in the submitters list, but the user account associated with the submission can't be modified.`: We should discuss this on a new issue. Maybe we auto-complete to user accounts underneath the form? The choices are not trivial.
* This seems reasonable for the short term, but we need a backup strategy now. Maybe we just use MongoDB Atlas's inbuilt tools?
* `I'll need a list of the first and last names, user ids, and emails as they are shown on the user admin table.` Assign me on this one.

**Comment by cesarvarela** [comment]
> The review form renders twice before it is stable

I don't know what you mean by the review form.

Good to merge this to staging?

@smcgregor

**Comment by smcgregor** [comment]
Yes.

If go to the submission review page, it will render, then it will update many of the badges.

**Pull request merged**